js/client.js

```diff
@@ -1325,8 +1325,13 @@
 				if (this.curSideRoom === oldRoom) this.curSideRoom = room;
 				if (this.sideRoom === oldRoom) this.sideRoom = room;
 			}
-			if (type === BattleRoom) this.roomList.push(room);
-			if (type === ChatRoom || type === BattlesRoom) this.sideRoomList.push(room);
+			if ([MainMenuRoom, TeambuilderRoom, LadderRoom, RoomsRoom].indexOf(type) < 0) {
```

Comment:
Right, sorry, I just adopted that from the previous code without thinking about it. Changed.